src/main/java/core/basesyntax/strategy/impl/PurchaseOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.strategy.OperationHandler;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(String fruit, int quantity, Map<String, Integer> storage) {
+        int currentQuantity = storage.getOrDefault(fruit, 0);
+        int updatedQuantity = currentQuantity - quantity;
+
+        if (updatedQuantity < 0) {
+            throw new RuntimeException("Not enough " + fruit + " in storage");
+        }
+
+        storage.put(fruit, updatedQuantity);

```

Comment:
Handling Purchase operation.
Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative.

You are correctly throwing a RuntimeException when the updatedQuantity is negative, which is good. However, it's important to include the attempted purchase quantity and the current available quantity in the exception message to provide more context to the exception.